### PR TITLE
Show suits tooltips

### LINKED ISSUE #2073: In-game explanation of the used variant
If I'm a new player, and started e.g. spectating a game of some variant, then I'll have no idea what is this "light pink" or "muddy rainbow" suit is. So I'd move the mouse to that suit, and there would be a tooltip that explains that "light pink is ...."

And such tooltip should mention which letter I should use in the notes to mark my card as of that suit, like "3y" (yellow 3) but for some other suit. Such letter is currently only visible in the colorblind mode.

### PR DESCRIPTION
**Pull request by DarthGandalf**
Fix #2073

### CODE REVIEW

File: client/src/game/types/CardState.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import Color from "./Color";
 
+// "playStack" is an empty play stack, "deck" is a play stack with some cards on it
```
Comment (Zamiell) [inline client/src/game/types/CardState.ts:3]: its not necessarily empty

**Comment by Zamiell** [comment]
fails CI, make it pass CI

**Comment by Zamiell** [comment]
does this only affect spectators of ongoing games?

File: client/src/game/types/CardState.ts (hunk 1 above)
Comment (DarthGandalf) [reply client/src/game/types/CardState.ts:3]: Then why there is a need for two?

**Comment by DarthGandalf** [comment]
No, it works for players too

**Comment by Zamiell** [comment]
that sounds very distracting for players, to have tooltips pop up constantly as they hover over cards when considering to play them

File: client/src/game/types/CardState.ts (hunk 1 above)
Comment (Zamiell) [reply client/src/game/types/CardState.ts:3]: what do you mean two?

Comment (DarthGandalf) [reply client/src/game/types/CardState.ts:3]: What do these values mean? In my tests I needed to handle both: one for case of empty stack, another for case of something on it.
What's the intended difference?

**Comment by DarthGandalf** [comment]
> that sounds very distracting for players, to have tooltips pop up constantly as they hover over cards when considering to play them

Do you prefer these tolltips somewhere else instead of on stacks?

**Comment by DarthGandalf** [comment]
> as they hover over cards when considering to play them

How do you play the card which is played already? The tooltip is not shown for cards in hands

**Comment by Zamiell** [comment]
```
  isOnPlayStack(): boolean {
    const { location } = this.state;
    return location === "playStack" || location === "deck";
  }
  ```
i dont understand this function, if the location is in the deck then why would it be on a pack stack?

**Comment by Zamiell** [comment]
> Do you prefer these tolltips somewhere else instead of on stacks?

do they appear on the cards or just on an empty stack?

**Comment by DarthGandalf** [comment]
> i dont understand this function, if the location is in the deck then why would it be on a play stack?

I'm wondering the same. I don't understand what exactly "deck" means here. From what `state.location` was returning, we need these 2 values here. One is for the empty play stack, another for a play stack with some cards on it.

**Comment by DarthGandalf** [comment]
This is why I had to add a comment to the enum.

**Comment by Zamiell** [comment]
i think you uncovered a real bug in the website then
we will need to investigate why it says "deck" if the real  location is the play stack
that function "isOnPlayStack()" should be deleted and we should fix this other bug first

**Comment by Zamiell** [comment]
can you look into this?

**Comment by DarthGandalf** [comment]
It probably is set in src/game/reducers/initialStates/initialCardState.ts and never updated?

I don't know what this code should do and how the website works in general.

**Comment by DarthGandalf** [comment]
I removed the function. Now the tooltip is shown only on non-empty stacks.

I'll leave it to you to fix the bug how you see fit.

**Comment by Zamiell** [comment]
looks good now thanks

**Pull request merged**